crates/bevy_reflect/bevy_reflect_derive/src/lib.rs

```diff
@@ -575,6 +575,155 @@ pub fn impl_reflect_value(input: TokenStream) -> TokenStream {
     )
 }
 
+#[proc_macro]
+pub fn impl_reflect_struct_and_from_reflect_struct(input: TokenStream) -> TokenStream {
+    let mut iter = input.into_iter().peekable();
+    let (r#constructor, ctor) = (iter.next().unwrap(), iter.next().unwrap());
+    match r#constructor {
+        proc_macro::TokenTree::Ident(i) => {
+            if i.to_string() != "Constructor" {
+                panic!("Invalid constructor syntax")
```

Comment:
This (and the other panics) should probably return a compile error instead.

Reply:
How is that done? Make it emit an `error!` call?

Reply:
I like to create a [`syn::Error`](https://docs.rs/syn/latest/syn/parse/struct.Error.html) and then return `my_error.into_compile_error()`.

The great thing about `syn::Error` is that it allows you to [combine](https://docs.rs/syn/latest/syn/parse/struct.Error.html#method.combine) multiple errors into one, which allows the compiler to give the most amount of information back to the user.